common/events/player_event_logs.cpp

```diff
@@ -123,21 +134,466 @@ void PlayerEventLogs::ProcessBatchQueue()
+
+				{
+					std::stringstream ss;
+					ss << r.event_data;
+					cereal::JSONInputArchive ar(ss);
+					in.serialize(ar);
+				}
+
+				out.npc_id          = in.npc_id;
+				out.npc_name        = in.npc_name;
+				out.handin_copper   = in.handin_money.copper;
+				out.handin_silver   = in.handin_money.silver;
+				out.handin_gold     = in.handin_money.gold;
+				out.handin_platinum = in.handin_money.platinum;
+				out.return_copper   = in.return_money.copper;
+				out.return_silver   = in.return_money.silver;
+				out.return_gold     = in.return_money.gold;
+				out.return_platinum = in.return_money.platinum;
+				out.is_quest_handin = in.is_quest_handin;
+				out.created_at      = r.created_at;
+
+				if (m_etl_settings.contains(PlayerEvent::EventType::NPC_HANDIN)) {
+					r.etl_table_id = m_etl_settings.at(PlayerEvent::EventType::NPC_HANDIN).next_id;
+					m_etl_settings.at(PlayerEvent::EventType::NPC_HANDIN).next_id++;
+				}
+
+				if (!in.handin_items.empty()) {
+					for (auto const &i: in.handin_items) {
+						out_entries.charges                    = i.charges;
+						out_entries.evolve_amount              = 0;
+						out_entries.evolve_level               = 0;
+						out_entries.item_id                    = i.item_id;
+						out_entries.player_event_npc_handin_id = r.etl_table_id;
+						out_entries.type                       = 1;
+						out.created_at                         = r.created_at;
+
+						if (!i.augment_ids.empty()) {
+							uint32 augments[6]{};
+							for (int x = 0; x < i.augment_ids.size(); x++) {
+								augments[x] = i.augment_ids[x];
+							}
+							out_entries.augment_1_id = augments[0];
+							out_entries.augment_2_id = augments[1];
+							out_entries.augment_3_id = augments[2];
+							out_entries.augment_4_id = augments[3];
+							out_entries.augment_5_id = augments[4];
+							out_entries.augment_6_id = augments[5];
+						}
+					etl_queues.npc_handin_entries.push_back(out_entries);
+					}
+				}
+
+				if (!in.return_items.empty()) {
+					for (auto const &i: in.handin_items) {
+						out_entries.charges                    = i.charges;
+						out_entries.evolve_amount              = 0;
+						out_entries.evolve_level               = 0;
+						out_entries.item_id                    = i.item_id;
+						out_entries.player_event_npc_handin_id = r.etl_table_id;
+						out_entries.type                       = 2;
+						out.created_at                         = r.created_at;
+
+						if (!i.augment_ids.empty()) {
+							uint32 augments[6]{};
+							for (int x = 0; x < i.augment_ids.size(); x++) {
+								augments[x] = i.augment_ids[x];
+							}
+							out_entries.augment_1_id = augments[0];
+							out_entries.augment_2_id = augments[1];
+							out_entries.augment_3_id = augments[2];
+							out_entries.augment_4_id = augments[3];
+							out_entries.augment_5_id = augments[4];
+							out_entries.augment_6_id = augments[5];
+						}
+						etl_queues.npc_handin_entries.push_back(out_entries);
```

Comment:
`.reserve()` to pre-allocate memory since we know the size ahead of time

Reply:
Added reserve to queue.